libraries/erlang_install.rb

```diff
@@ -0,0 +1,83 @@
+#
+# Cookbook Name:: languages
+# HWRP:: erlang_install
+#
+# Copyright 2015, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class Chef
+  class Resource::ErlangInstall < Resource::LWRPBase
+    resource_name :erlang_install
+
+    actions :install
+    default_action :install
+
+    attribute :version,  kind_of: String, name_attribute: true
+    attribute :checksum, kind_of: String, required: true
```

Comment:
What is the purpose of this attribute/property? It seems we can/should just check the desired version is installed. We could do that a few ways:

* Check for the install directory like we do in `ruby_install`: https://github.com/chef-cookbooks/omnibus/blob/master/libraries/ruby_install.rb#L89-L96
* Dump the version with `erl`, so do something like this:
```
$ /opt/delivery/embedded/bin/erl -noshell -eval 'io:fwrite("~s\n", [erlang:system_info(otp_release)]).' -s erlang halt
17
```
* Read the `OTP_VERSION` file:
```
$ cat /opt/delivery/embedded/lib/erlang/releases/17/OTP_VERSION
17.5
```